### DISCUSSION TITLE
How to configure 5 rotation axes to control a robot arm.

### DISCUSSION THREAD

**Discussion by GrblGru** [fra589/grbl-Mega-5X#338] [Q&A]
A GrblGru user wants to control his robot arm.
For this purpose 0 linear axes and 5 rotational axes are needed.
Is it possible to configure this ?

**Comment by fra589** [comment]
hi @GrblGru,  
   
Why wouldn't that be possible?

```
#define N_AXIS 5
#define N_AXIS_LINEAR 0
```

In this case, axis names have no importance as long as they are included in the list of authorized names and correct step per unit, limits and acceleration parameters.  
  
@++;  
Gauthier.

**Comment by GrblGru** [comment]
Hi Gauthier,

thank you very much, with 5 axes it works great. :)

Because it works so well, I also tried it again with 6 axes.
So: #define N_AXIS 6 // Number of axes (3 to 6)
I used A,B,C,D,E and H as axis names.
When I try to move the H-axis, it looks like the interface crashes immediately. All other axes work fine. Of course this can be caused by my program, but I can't find anything.

So my question is if this setting is allowed at all and if it **should** work this way.

Thanks for your help :)
GrblGru

**Reply by fra589** [reply]
Hi @GrblGru, 

This setting should work, but...

Since I don't need it myself, I wrote the 6-axis version for someone else's needs. I Really only tried the 6 axis version very few time, only for debugging with axis letters X, Y, Z, A, B and C.

Moreover, the functionalities that I added since that may have added conflicts on certain specifics configurations.

Can you test the 6 axis functionality with axes names X, Y, Z, A, B and C th see if your problem come with the 6t axis or with the name H?

@++;  
Gauthier.

**Comment by GrblGru** [comment]
Yes, I will be happy to do that, but for now I have a vacation :)
I will report when I'm back and have tried it.

Thanks for your help

**Comment by fra589** [comment]
Hi @GrblGru,  
  
Having a little free time, I just tested on a test card with XYZABC and ABCDEH, 6 axis configurations N_AXIS=6 and N_AXIS_LINEAR=0, and they they both work fine.

I wish you good vacations!

@++;
Gauthier.
